### ISSUE TITLE
Mention JSComponent in Explanation/Components docs?

### ISSUE THREAD

**Issue by cdeil** [holoviz/panel#7122]
@MarcSkovMadsen and @philippjfr - would it make sense to mention JSComponent or whatever the new ESM component classes are in the docs also under Explanation -> Components?

I wanted to learn about them and my first instinct was to look here:

https://holoviz-dev.github.io/panel/explanation/index.html#components

<img width="1350" alt="Screenshot 2024-08-10 at 21 47 18" src="https://github.com/user-attachments/assets/6d33127d-710f-4f15-9874-07240ba57f00">

**Comment by cdeil** [comment]
There's also an inconsistency where docs pages link to `reference/custom` but currently the files are generated under `reference/custom_components`.

Which one should it be?

https://holoviz-dev.github.io/panel/tutorials/expert/custom_components.html

<img width="1473" alt="Screenshot 2024-08-10 at 23 25 16" src="https://github.com/user-attachments/assets/30b3a6a5-7916-41ab-b6d7-24840f358673">

**Comment by MarcSkovMadsen** [comment]
The latter

**Comment by cdeil** [comment]
@MarcSkovMadsen - can you fix this as part of #7127 ?
(I could send a PR, but don't want to overlap)

One more thought: suggest to remove the "Prerequisites" from https://holoviz-dev.github.io/panel/tutorials/expert/custom_components.html#prerequisites 

I think like most other tutorials we can simply assume that Panel install is in place.
<img width="1277" alt="Screenshot 2024-08-11 at 19 11 26" src="https://github.com/user-attachments/assets/91bbeddf-10c4-4f6b-a3b1-db8c56d6d6a5">

**Comment by philippjfr** [comment]
>I think like most other tutorials we can simply assume that Panel install is in place.

Indeed, I'd be happy with a little admonition reminding people that `watchfiles` will give them a much better development experience.

**Comment by cdeil** [comment]
> reminding people that watchfiles will give them a much better development experience

@philippjfr - Or just make watchfiles a required dependency? ;-)

* Pro: everyone gets good dev experience always, no need for reminders.
* Con: an extra dependency. But it's a small (ca 1 MB) and simple one compared to pandas & bokeh and other core dependencies panel has already.

**Comment by MarcSkovMadsen** [comment]
The basic Project tutorials are made as self contained as possible.

We share links to them on social media. They can the first docs a new user meets. So I believe they should be there. But who knows What correct approach is?

**Comment by cdeil** [comment]
> The basic Project tutorials are made as self contained as possible.

It's currently under "Expert tutorials". I guess for people that know JS it could be considered and go under "Basic" or "Intermediate".
